frontend/test/framework/CMakeLists.txt

```diff
@@ -21,6 +21,7 @@ set_tests_properties(testRecursionFails PROPERTIES WILL_FAIL TRUE)
 comp_unit_test(testDependencies)
 comp_unit_test(testErrorTracking)
 comp_unit_test(testIds)
+comp_unit_test(testQueryEquivalence)
```

Comment:
I'd think that this is a resolution test rather than a framework test because it's focused on MostSpecificCandidate ? Maybe it should be called testMostSpecificCandidateQueries or something?

Reply:
I envision this file being populated with other data structures which we want to ensure behave properly under the query system's constraints. I've only tested the data structures affected by my PR, but I hope to make use of it for other DSs.